src/index.js

```diff
@@ -63,32 +51,33 @@ module.exports = function module (moduleOptions) {
       staticRoutes = staticRoutes.filter(route => minimatch.match(route))
     })
 
-    if (this.options.dev || options.generate) {
+    if (this.options.dev) {
       // Create a cache for routes
       cache = createCache(staticRoutes, options)
-    }
-
-    if (!this.options.dev) {
-      // TODO on build process only
+    } else {
       // Save static routes
       fs.ensureDirSync(path.resolve(this.options.buildDir, 'dist'))
       fs.writeJsonSync(jsonStaticRoutesPath, staticRoutes)
+    }
+  })
 
-      // TODO on generate process only and not on build process
-      if (options.generate) {
-        (async () => {
-          // Generate static sitemap.xml
-          const routes = await cache.get('routes')
-          const sitemap = await createSitemap(options, routes)
-          const xml = await sitemap.toXML()
-          await fs.ensureFile(xmlGeneratePath)
-          await fs.writeFile(xmlGeneratePath, xml)
-          if (options.gzip) {
-            const gzip = await sitemap.toGzip()
-            await fs.writeFile(gzipGeneratePath, gzip)
-          }
-        })()
-      }
+  if (options.generate) {
+    consola.warn('The option `generate` isn\'t needed anymore')
```

Comment:
```suggestion
    consola.warn('The option `sitemap.generate` isn\'t needed anymore')
```

Reply:
So the user knows which prop it is exactly

Reply:
Ready! :+1: